elasticsearch/helpers/vectorstore/_async/vectorstore.py

```diff
@@ -344,8 +344,8 @@ async def _create_index_if_not_exists(self) -> None:
     async def max_marginal_relevance_search(
         self,
         *,
-        embedding_service: AsyncEmbeddingService,
```

Comment:
Oh, I did not realize this was not useful. Unfortunately, you can't remove it as the LangChain integration passes this parameter: https://github.com/langchain-ai/langchain-elastic/blob/9915f1e16d20e992896a651f7a3ec94db56a76f0/libs/elasticsearch/langchain_elasticsearch/vectorstores.py#L816-L825

Reply:
Okay I will add it as an optional param, my plan was to create another PR in the langchain repo.